internal/engine/configs/configs_controller.go

```diff
@@ -92,6 +93,10 @@ func (cc *ConfigsController) needsBuild(ctx context.Context, st store.RStore) (b
 		reason = reason.With(model.BuildReasonFlagTiltfileArgs)
 	}
 
+	if state.PendingTiltfileTrigger.After(lastStartTime) {
```

Comment:
this needs to use AfterOrEqual, like other places in the build engine. the test failure on windows are legit; they're happening because the windows system clock isn't monotonic in the way the unix clock is